addons/payment/models/payment_acquirer.py

```diff
@@ -722,28 +722,64 @@ def _log_payment_transaction_received(self):
     @api.multi
     def _set_transaction_pending(self):
         '''Move the transaction to the pending state(e.g. Wire Transfer).'''
-        if any(trans.state != 'draft' for trans in self):
-            raise ValidationError(_('Only draft transaction can be processed.'))
-
-        self.write({'state': 'pending', 'date': datetime.now().strftime(DEFAULT_SERVER_DATETIME_FORMAT)})
-        self._log_payment_transaction_received()
+        def filter_trans(trans):
+            if trans.state in ('pending',):
+                _logger.info('Trying to set to pending a transaction already pending tx (ref: %s, actual state: %s, required state: %s)' % (trans.reference, trans.state, 'pending'))
+                return False
+
+            if trans.state not in ('draft',):
+                _logger.warning('Only draft transaction can be set to pending (ref: %s, actual state: %s, required state: %s)' % (trans.reference, trans.state, 'pending'))
+                return False
+            return True
+
+        tx_to_process = self.filtered(filter_trans)
```

Comment:
I really dislike that your filtering method logs stuff... for me, it should only handle the recordset and the logging should be done in the flow
``` python
tx_to_process = self.filtered(lambda tx: tx.state == 'draft')
tx_already_pending = self.filtered(lambda tx: tx.state == 'pending')
tx_bad_state = self - tx_to_process - tx_already_pending
for pending_tx in tx_already_pending:
  _logger.info('blablabla')
for bad_tx in tx_bad_state:
  _logger.warning('machinmachinmachin')
# then go one with the tx_to_process recordset
```

Reply:
same comment applies for all methods that use this